Order form

jessieengstrom opened the pull request:
- added an OrderForm component
- added an OrderPage component
- added state to OrderPage (not sure if this is the correct place for it?)
- added two methods to OrderPage (handleSubmit and handleChange)
-ccurrently logging to the console the current state after the order is submitted

a15n commented on src/components/OrderPage.js:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    handleSubmit(e) {
+        e.preventDefault();
```
nice

a15n commented on src/components/OrderPage.js:
```diff
@@ -0,0 +1,43 @@
+        const target = e.target
+        const name = target.name
+        const value = target.value
+        this.setState(state => ({[name] : value}))
```
nice

a15n commented on src/components/OrderPage.js:
```diff
@@ -0,0 +1,43 @@
+    }
+
+    handleChange(e) {
+        const target = e.target
```
good use for es6 object destructuring

https://wesbos.com/destructuring-objects/

a15n commented on src/components/OrderPage.js:
```diff
@@ -0,0 +1,43 @@
+
+    render() {
+        return (
+            <div>
```
div not necessary

a15n commented on src/components/OrderForm.js:
```diff
@@ -0,0 +1,84 @@
+                    <input 
+                        type='text'
+                        name='escrowNumber'
+                        onChange={props.handleChange}/>
```
personal preference but let's always have elements open and close on the same line
```
<input
   ... some stuff
/>
```

a15n commented on src/components/OrderForm.js:
```diff
@@ -0,0 +1,84 @@
+
+export default function OrderForm(props) {
+    return (
+        <div>
```
you are right that a React component can't return multiple elements. Wrapping multiple elements in a div works, but there is another way. Wrapping them in a Fragment. This removes an unnecessary div from the DOM and makes the page slightly lighter and faster.

https://reactjs.org/docs/fragments.html

a15n commented on src/components/OrderForm.js:
```diff
@@ -0,0 +1,84 @@
+import React from 'react';
+
+export default function OrderForm(props) {
```
biggest piece of feedback is to move all of the logic inside of the OrderForm component, instead of passing it down to the OrderForm via props. We should keep logic as close to its source as possible.

jessieengstrom commented:
Made the following changes:
- moved handleSubmit and handleChange methods to the OrderForm component
- removed an unnecessary <div> 
- used <React.Fragment> instead of <div>
